src/Agent.Worker/Release/Artifacts/BuildArtifact.cs

```diff
@@ -181,26 +184,35 @@ private bool Match(ServerBuildArtifact buildArtifact, ArtifactDefinition artifac
                 var fileShareArtifact = new FileShareArtifact();
                 await fileShareArtifact.DownloadArtifactAsync(executionContext, HostContext, artifactDefinition, fileShare, downloadFolderPath);
             }
-            else if (string.Equals(buildArtifact.Resource.Type, WellKnownArtifactResourceTypes.Container, StringComparison.OrdinalIgnoreCase))
+            else if (buildArtifactDetails != null &
+                     string.Equals(buildArtifact.Resource.Type, WellKnownArtifactResourceTypes.Container, StringComparison.OrdinalIgnoreCase))
             {
-                executionContext.Output("Artifact Type: ServerDrop");
-
-                // TODO:Get VssBinFetchclient and get away from zipstream downloader
-                Stream contentStream;
+                string[] parts = buildArtifact.Resource.Data.Split(new[] { '/' }, 3);
```

Comment:
Can you add a comment explain what are the 3 parts, also, will the Data.Split throw? after split, make sure its length=3, since you call parts[1]/[2] directly later.